x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/results/StreamingUnifiedChatCompletionResults.java

```diff
@@ -60,76 +61,83 @@ public boolean isStreaming() {
     }
 
     @Override
-    public List<? extends InferenceResults> transformToCoordinationFormat() {
+    public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
         throw new UnsupportedOperationException("Not implemented");
     }
 
-    @Override
-    public List<? extends InferenceResults> transformToLegacyFormat() {
-        throw new UnsupportedOperationException("Not implemented");
-    }
+    public record Results(Deque<ChatCompletionChunk> chunks) implements InferenceServiceResults.Result {
+        public static String NAME = "streaming_unified_chat_completion_results";
 
-    @Override
-    public Map<String, Object> asMap() {
-        throw new UnsupportedOperationException("Not implemented");
-    }
-
-    @Override
-    public String getWriteableName() {
-        throw new UnsupportedOperationException("Not implemented");
-    }
-
-    @Override
-    public void writeTo(StreamOutput out) throws IOException {
-        throw new UnsupportedOperationException("Not implemented");
-    }
+        public Results(StreamInput in) throws IOException {
+            this(deque(in));
+        }
 
-    @Override
-    public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
-        throw new UnsupportedOperationException("Not implemented");
-    }
+        private static Deque<ChatCompletionChunk> deque(StreamInput in) throws IOException {
+            return in.readCollection(ArrayDeque::new, ((stream, deque) -> deque.offer(new ChatCompletionChunk(stream))));
+        }
 
-    public record Results(Deque<ChatCompletionChunk> chunks) implements ChunkedToXContent {
         @Override
         public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
             return Iterators.concat(Iterators.flatMap(chunks.iterator(), c -> c.toXContentChunked(params)));
         }
-    }
 
-    public static class ChatCompletionChunk implements ChunkedToXContent {
-        private final String id;
+        @Override
+        public String getWriteableName() {
+            return NAME;
+        }
 
-        public String getId() {
-            return id;
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeCollection(chunks, StreamOutput::writeWriteable);
         }
 
-        public List<Choice> getChoices() {
-            return choices;
+        @Override
+        public boolean equals(Object o) {
+            if (o == null || getClass() != o.getClass()) return false;
+            Results results = (Results) o;
+            return dequeEquals(chunks, results.chunks());
         }
 
-        public String getModel() {
-            return model;
+        private static boolean dequeEquals(Deque<?> thisDeque, Deque<?> otherDeque) {
```

Comment:
Looks like we use this in a couple places, do you think it'd be worth moving it to a utility function?